### PR TITLE
MueLu: Add test for aggregation export

### PR DESCRIPTION
**Pull request by pohm01**
@trilinos/muelu 

## Motivation
Add testing to the vtu output files of aggregation export factory.
Related to #9024 

## Testing
The visualization of the aggregates makes sense.

### CODE REVIEW

File: packages/muelu/test/viz/AggExport.cpp [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+
+#ifdef HAVE_MUELU_BELOS
+#include <BelosConfigDefs.hpp>
+#include <BelosLinearProblem.hpp>
+#include <BelosBlockCGSolMgr.hpp>
+#include <BelosPseudoBlockCGSolMgr.hpp>
+#include <BelosBlockGmresSolMgr.hpp>
+#include <BelosXpetraAdapter.hpp>     // => This header defines Belos::XpetraOp
+#include <BelosMueLuAdapter.hpp>      // => This header defines Belos::MueLuOp
+#endif
```
Comment (mayrmt) [inline packages/muelu/test/viz/AggExport.cpp:81]: This driver does not use `Belos`, so these should be removed.
```suggestion
```

File: packages/muelu/test/viz/AggExport.cpp [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+    tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("ScalingTest: 1.5 - MueLu read XML")));
+
+    RCP<HierarchyManager> mueLuFactory = rcp(new ParameterListInterpreter(paramList));
+
+    comm->barrier();
+    tm.reset();
+    tm = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("ScalingTest: 2 - MueLu Setup")));
+
+    RCP<Hierarchy> H;
+    for (int i = 0; i <= numRebuilds; i++) {
```
Comment (mayrmt) [inline packages/muelu/test/viz/AggExport.cpp:357]: What's the purpose of rebuilding the hierarchy here? If there's none, this loop can be removed. If so, please also remove the command line argument to control the number of rebuilds.

File: packages/muelu/test/viz/AggExport.cpp [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+    std::string domainMapFile;                         clp.setOption("domainmap",             &domainMapFile,     "domainmap data file");
+    std::string rangeMapFile;                          clp.setOption("rangemap",              &rangeMapFile,      "rangemap data file");
+    std::string matrixFile;                            clp.setOption("matrix",                &matrixFile,        "matrix data file");
+    std::string coordFile;                             clp.setOption("coords",                &coordFile,         "coordinates data file");
+    std::string nullFile;                              clp.setOption("nullspace",             &nullFile,          "nullspace data file");
+    int         numRebuilds       = 0;                 clp.setOption("rebuild",               &numRebuilds,       "#times to rebuild hierarchy");
+    int         maxIts            = 200;               clp.setOption("its",                   &maxIts,            "maximum number of solver iterations");
+    bool        scaleResidualHist = true;              clp.setOption("scale", "noscale",      &scaleResidualHist, "scaled Krylov residual history");
```
Comment (mayrmt) [inline packages/muelu/test/viz/AggExport.cpp:161]: Unused, so please remove:
```suggestion
```

File: packages/muelu/test/viz/AggExport.cpp [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+    // =========================================================================
+    // Parameters initialization
+    // =========================================================================
+    GO nx = 100, ny = 100, nz = 100;
+    Galeri::Xpetra::Parameters<GO> galeriParameters(clp, nx, ny, nz, "Laplace2D"); // manage parameters of the test case
+    Xpetra::Parameters             xpetraParameters(clp);                          // manage parameters of Xpetra
+
+    std::string xmlFileName       = "viztest.xml";     clp.setOption("xml",                   &xmlFileName,       "read parameters from a file");
+    bool        printTimings      = true;              clp.setOption("timings", "notimings",  &printTimings,      "print timings to screen");
+    int         writeMatricesOPT  = -2;                clp.setOption("write",                 &writeMatricesOPT,  "write matrices to file (-1 means all; i>=0 means level i)");
```
Comment (mayrmt) [inline packages/muelu/test/viz/AggExport.cpp:150]: Unused, so please remove:
```suggestion
```

**Review by mayrmt — CHANGES_REQUESTED** [review]
@pohm01  Thanks for adding these tests! This will really help to protect this functionality. There have been problems with aggregate visualization in the past, but hopefully that's fixed and guarded against future changes now.

I made some comments related to cleanup of the test driver, just to make it easier to understand for others.

**Review by jhux2 — DISMISSED** [review]
lgtm

**Review by mayrmt — APPROVED** [review]
@pohm01 Thanks for the cleanup!

**Review by mayrmt — APPROVED** [review]

**Review by jhux2 — APPROVED** [review]
lgtm

**Pull request merged**